### PR TITLE
fix(node-runtime-worker-thread): Re-enable tests and fix the leak COMPASS-4614

### PR DESCRIPTION
**Pull request by gribnoysup**
This PR fixes a Worker leak caused by sinon spies and adds a workaround for possible Node.js issue that causes Worker MessagePort to keep the process running even after all tests are finished

Some investigation details are in [this thread](https://mongodb.slack.com/archives/GJ2CBBY86/p1613032460019700)

### CODE REVIEW

File: packages/node-runtime-worker-thread/src/worker-runtime.spec.ts [hunk 1]
```diff
@@ -50,7 +49,14 @@ describe.skip('worker', () => {
 
   afterEach(async() => {
     if (worker) {
-      await worker.terminate();
+      // Possibly a Node.js bug that causes worker process to still be ref-ed
+      // after termination. To work around that, we are unrefing worker manually
+      // *immediately() after terminate method is called even though it should
```
Comment (addaleax) [inline packages/node-runtime-worker-thread/src/worker-runtime.spec.ts:54]:
```suggestion
      // *immediately()* after terminate method is called even though it should
```

(I guess?)

Also, if you want to, feel free to add a reference to [https://github.com/nodejs/node/pull/37319](https://tinyurl.com/1bgd7wlb) here :)

**Review by addaleax — APPROVED** [review]

File: packages/node-runtime-worker-thread/src/worker-runtime.spec.ts (hunk 1 above)
Comment (gribnoysup) [reply packages/node-runtime-worker-thread/src/worker-runtime.spec.ts:54]: Oops, yeah, thanks! And totally will add a link. Wow, you fixed it in no time at all!

Comment (addaleax) [reply packages/node-runtime-worker-thread/src/worker-runtime.spec.ts:54]: I mean, you already did the hard part here :)

**Comment by gribnoysup** [comment]
Ahhh, green at last!

**Pull request merged**